_source/_posts/2022-10-01-spring-vault.md

```diff
@@ -0,0 +1,639 @@
+
+```shell
+auth0 login
+```
+
+The terminal will display a device confirmation code and open a browser session to activate the device.
+
+**NOTE**: My browser was not displaying anything, so I had to manually activate the device by opening the URL https://auth0.auth0.com/activate?user_code={deviceCode}
+
+On successful login, you will see the tenant, which you will use as issuer later:
+
+```shell
+✪ Welcome to the Auth0 CLI 🎊
+
+If you don't have an account, please create one here: https://auth0.com/signup.
+
+Your device confirmation code is: KGFL-LNVB
+
+ ▸    Press Enter to open the browser to log in or ^C to quit...
+
+Waiting for the login to complete in the browser... ⣻Opening in existing browser session.
+Waiting for the login to complete in the browser... done
+
+ ▸    Successfully logged in.
+ ▸    Tenant: dev-avup2laz.us.auth0.com
+```
+
+The next step is to create a client app:
+
+```shell
+auth0 apps create
+```
+
+When prompted, choose the following options:
+
+- Name: `vault-demo-app-auth0`
+- Description: `<optional description>`
+- Type: `Regular Web Application`
+- Callback URLs: `http://localhost:8080/login/oauth2/code/auth0`
+- Logout URLs: `http://localhost:8080`
+
+Once the app is created, you will see the OIDC app's configuration:
+
+```shell
+Name: vault-demo-app-auth0
+Description: Demo project of a Spring Boot application with Vault protected secrets
+Type: Regular Web Application
+Callback URLs: http://localhost:8080/login/oauth2/code/auth0
+Allowed Logout URLs: http://localhost:8080
+
+=== dev-avup2laz.us.auth0.com application created
+
+ CLIENT ID            ****
+ NAME                 vault-demo-app-auth0
+ DESCRIPTION          Demo project of a Spring Boot application with Vault protected secrets
+ TYPE                 Regular Web Application
+ CALLBACKS            http://localhost:8080/login/oauth2/code/auth0
+ ALLOWED LOGOUT URLS  http://localhost:8080
+ ALLOWED ORIGINS
+ ALLOWED WEB ORIGINS
+ TOKEN ENDPOINT AUTH
+ GRANTS               implicit, authorization_code, refresh_token, client_credentials
+
+▸    Quickstarts: https://auth0.com/docs/quickstart/webapp
+▸    Hint: Test this app's login box with 'auth0 test login ****'
+▸    Hint: You might wanna try 'auth0 quickstarts download ****
+```
+
+**NOTE**: The client secret is [not displayed](https://github.com/auth0/auth0-cli/issues/488) in the CLI output for regular applications. You can run `auth0 apps open` to open a browser in the application configuration. Then copy the secret from the **Settings** tab.
+
+You can now run the demo app passing the oidc configuration through environment variables:
+
+```shell
+SPRING_SECURITY_OAUTH2_CLIENT_PROVIDER_AUTH0_ISSUER_URI=https://{tenant}/ \
+SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_AUTH0_CLIENT_ID={clientId} \
+SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_AUTH0_CLIENT_SECRET={clientSecret} \
+./mvnw spring-boot:run
```

Comment:
I get a rate limit error when I try to log in. I believe this is because no scopes are defined.

Reply:
It did not happen to me. I commited your suggestions and the scopes are there now.